plugins/modules/slack.py

```diff
@@ -469,6 +667,20 @@ def main():
     blocks = module.params['blocks']
     message_id = module.params['message_id']
     prepend_hash = module.params['prepend_hash']
+    upload_file = module.params["upload_file"]
+
+    if upload_file:
+        try:
+            upload_response = upload_file_to_slack(
```

Comment:
Hmm, this seems to be very separate from the rest of the module. Maybe this should be a new module that only does file uploading, instead of squeezing this into the current module which is about posting messages?